mlir/test/builder-tests/conditional.mlir

```diff
@@ -0,0 +1,33 @@
+// RUN: hugr-mlir-opt %s | FileCheck %s
+
+!unit = tuple<>
+!bool = !hugr.sum<!unit,!unit>
+
+hugr.module @basic_conditional {
+    hugr.func @foo [](%p : !bool) -> index {
+        %i = conditional(%p : !bool) -> index
+        output %i : index
+    }
+}
+
+// CHECK-LABEL: hugr.module @basic_conditional
+// CHECK: func
+// CHECK: = conditional({{.*}} : !hugr.sum<tuple<>, tuple<>>) -> index
+// CHECK: output %
+
+hugr.module @basic_conditional_module {
+    func @main[](%x: index) -> index {
+         const @c : !unit = #hugr.unit
```

Comment:
We're not, it's unit!. I think I'm missing a `hugr.tag`, to construct a bool. This ir is invalid, but because of missing `SymbolUserOpInterface` it's not being caught yet. Will fix.